enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -761,6 +761,11 @@ func (mc *MigrationCache) copyDataInBackground() error {
 
 		select {
 		case <-mc.quitChan:
+			// Drain copy channel on shutdown
+			close(mc.copyChan)
```

Comment:
I think there's a small window where we might try to write on a closed channel. 
We start shutting down and concurrently there are still some remaining requests that have not called into the cache yet (or have already entered into the cache code but may take some small amount of time before scheduling a copy, for example during FIndMissing handling). If we close the channel here, the app will panic when we try to enqueue another copy.

You could drain the queue by reading from the channel until there's nothing left in the queue. We might drop some last second copies, but I think it's ok if this is best effort.

Reply:
Ah right - fixed